v2/internal/internalguantlettest/call_options.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package internalgauntlettest
+
+import (
+	"fmt"
+
+	"go.uber.org/multierr"
+	"go.uber.org/yarpc/v2"
+)
+
+// intended to be used with every request
+func newCallOptions(resHeaders *map[string]string) []yarpc.CallOption {
+	return []yarpc.CallOption{
+		yarpc.ResponseHeaders(resHeaders),
+		yarpc.WithHeader(_headerKeyReq, _headerValueReq),
+		yarpc.WithShardKey(_shardKey),
+		yarpc.WithRoutingKey(_routingKey),
+		yarpc.WithRoutingDelegate(_routingDelegate),
+	}
+}
+
+// intended to be invoked by an inbound handler
+func validateCallOptions(call *yarpc.Call, encoding yarpc.Encoding) error {
```

Comment:
Should we also display the expected attribute for all of these error messages?
```
expected caller (%q), actual (%q).
```
or
```
expected caller: %q, got: %q
```